### PR TITLE
upgrade setup-java action. Current version uses set-output

### PR DESCRIPTION
**Pull request by jburel**
see also https://github.com/ome/action-gradle/pull/1

Check that the build via GHA is green

### CODE REVIEW

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-push#354](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/354/). See the [console output](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/354/consoleText) for more details.
Possible conflicts:
  - PR #3815 sbesson 'Java 17 support'
    - .github/workflows/maven.yml

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-push#1503](https://merge-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/1503/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/1503/consoleText) for more details.
Possible conflicts:
  - PR #3815 sbesson 'Java 17 support'
    - .github/workflows/maven.yml

--conflicts

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -13,17 +13,18 @@ jobs:
...
       maven_commands: install # default is install
     steps:
       - uses: actions/checkout@v2
       - name: Set up JDK ${{ matrix.java }}
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v3
         with:
           java-version: ${{ matrix.java }}
+          distribution: 'zulu'
       - name: Cache Maven packages
```
Comment (melissalinkert) [inline .github/workflows/maven.yml:28]: Should this be replaced with `cache: 'maven'` in the `setup-java` step?

Comment (jburel) [reply .github/workflows/maven.yml:28]: yes. I forgot to do that in that repo.
Well spotted

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [BIOFORMATS-push#356](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/356/). See the [console output](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-push/356/consoleText) for more details.
Possible conflicts:
  - PR #3815 sbesson 'Java 17 support'
    - .github/workflows/maven.yml
    - .github/workflows/ant.yml
  - PR #3912 melissalinkert 'Upload one set of artifacts from the GitHub Actions ant build'
    - .github/workflows/ant.yml

--conflicts

**Review by dgault — APPROVED** [review]
Changes all look good to me and the builds via GitHub actions are green and all the steps look to be correct and running as expected.

**Pull request merged**